python/dataproc_templates/pubsublite/pubsublite_to_bigquery.py

```diff
@@ -0,0 +1,173 @@
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_PROJECT_ID}',
+            dest=constants.PUBSUBLITE_TO_BQ_PROJECT_ID,
+            required=True,
+            help='BQ Project ID'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_OUTPUT_DATASET}',
+            dest=constants.PUBSUBLITE_TO_BQ_OUTPUT_DATASET,
+            required=True,
+            help='BigQuery output dataset'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE}',
+            dest=constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE,
+            required=True,
+            help='BigQuery output table'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_BATCH_SIZE}',
+            dest=constants.PUBSUBLITE_TO_BQ_BATCH_SIZE,
+            required=False,
+            default=1000,
+            help='Number of records to be written per message to BigQuery'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET}',
+            dest=constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET,
+            required=True,
+            help='Temporary bucket for the Spark BigQuery connector'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_CHECKPOINT_LOCATION}',
+            dest=constants.PUBSUBLITE_CHECKPOINT_LOCATION,
+            required=True,
+            help='Temporary folder for checkpoint location'
+        )
+
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        logger: Logger = self.get_logger(spark=spark)
+
+        # Arguments
+        input_subscription: str = args[constants.PUBSUBLITE_TO_BQ_INPUT_SUBSCRIPTION]
+        output_project_id: str = args[constants.PUBSUBLITE_TO_BQ_PROJECT_ID]
+        output_dataset: str = args[constants.PUBSUBLITE_TO_BQ_OUTPUT_DATASET]
+        output_table: str = args[constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE]
+        pubsublite_checkpoint_location: str = args[constants.PUBSUBLITE_CHECKPOINT_LOCATION]
+        bq_temp_bucket: str = args[constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET]
+
+        logger.info(
+            "Starting Pubsublite to Bigquery spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+        # Set configuration to connect to Cassandra by overwriting the spark session
+        spark = (
+            SparkSession
+                .builder
+                .appName("read-app")
+                .master("yarn")
+                .getOrCreate())
+
+        # Read
+        input_data=(spark.readStream \
+            .format(constants.FORMAT_PUBSUBLITE) \
+            .option(f"{constants.FORMAT_PUBSUBLITE}.subscription",f"projects/617357862702/locations/us-west1/subscriptions/{input_subscription}",) \
```

Comment:
The project ID variable containing the pubsublite subscription should be used here instead of hardcoding the ID.

Reply:
Removed the hardcoded values